packages/agent/src/agent-loop.ts

```
@@ -162,89 +185,161 @@ async function streamAssistantResponse(
+        )
+
+        eventQueue.push({ type: 'AgentStatus', message: '' }) // Clear status on success
+        return { assistantMessage, toolCalls }
+    } catch (error: any) {
+        let errorMsg = typeof error.message === 'string' ? error.message : String(error)
+
+        eventQueue.push({ type: 'AgentStatus', message: '' })
+
+        // Extract original error if p-retry wrapped it
+        if (error.name === 'AbortError' && error.originalError) {
+            error = error.originalError
```

Biome (2.5.1)

[error] Reassigning a catch parameter is confusing.

(lint/suspicious/noCatchAssign)